src/protocols/secure_channel/CASESession.cpp

```diff
@@ -555,9 +555,10 @@ void CASESession::OnResponseTimeout(ExchangeContext * ec)
 
 void CASESession::AbortPendingEstablish(CHIP_ERROR err)
 {
+    SessionEstablishmentStage state = MapCASEStateToSessionEstablishmentStage(mState);
```

Comment:
Please add a comment that this needs to come before Clear, because Clear will reset mState.